scripts/sync-dashboards-and-alert-rules.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# Sync the dashboards  and alert rules from https://opendev.org/openstack/sunbeam-charms/src/branch/main/charms/openstack-exporter-k8s/src
+
+git remote add -f sunbeam https://opendev.org/openstack/sunbeam-charms.git
+git checkout -b staging-branch sunbeam/main
+git subtree split -P charms/openstack-exporter-k8s/src -b k8s-exporter
+git checkout k8s-exporter
+git rm -r $(ls | grep -vE '^grafana_dashboards$|^prometheus_alert_rules$')
+git commit -m "Remove unwanted files"
+git checkout main
+git subtree add -P src/sunbeam k8s-exporter --squash
```

Comment:
I'm not sure what the benefit of using a git subtree is here?  Could you explain why we're using git subtree instead of something like checking out the sunbeam-charms repo and copying the relevant files?  Unless there's a compelling reason, I don't like this method - it does weird things with the git history and it's hard to follow what's happening.  It's also a bit of a foot-gun having this work on the current checkout and main branch; running this accidentally could cause confusion.

Have we considered a workflow more like this (untested example only).  The local script and makefie target would do this:

```
git clone --depth 1 https://opendev.org/openstack/sunbeam-charms.git $tmpdir
rsync -av $tmpdir/charms/openstack-exporter-k8s/src/grafana_dashboards/ ./grafana_dashboards
# same rsync for alert_rules
```

Then the github workflow could:

- run that script
- if there are changes:
  - add and commit the change on a feature branch
  - open a PR with that new branch

If we wanted something fancy in the commit message we could link to upstream commits (or include the commit messages) for all commits that touched the dashboard/alert files since the last time the files were updated.

Reply:
We are considering using git subtree because we believe this approach might be more stable and bug-free when using git commands. And since we use --squash, we can keep the git history clean and customize the commit message.
This is just an exploration. We can definitely consider the workflow you suggested or stick with our current one if the majority prefers that.

Reply:
I agree with @samuelallan72 here, I do not see reason why we're adding sunbeam as remote and then using subtree. Simply copy would be more appropriate. 

However, I think that script should be removed, and we can use GitHub actions directly.

Reply:
I don't think we should use github actions directly for this - a separate script would be helpful for developers to run locally on their machines for testing.  The script should do all the file syncing logic.  It shouldn't make any commits or mess with git; just copy files from the sunbeam repo.

Reply:
We can have helper script for developer, but in CI I think we should avoid using it. If we go with script, I would say that shell script with git command + cp / rsync would be better option than python code.